dev/.gitignore

```diff
@@ -18,3 +18,6 @@
 # Python virtual environments for dev tools
 .venv*/
 
+# yum-based builds copy a lot of files there
+tasks/linux-packages/apache-arrow/yum/build/
```

Comment:
Hmm. This directory isn't created by default.
Did you use `BUILD_DIR=build`?

https://github.com/apache/arrow/blob/4565b613a488516b9bcbc664ad21d92051ea4fe6/dev/tasks/linux-packages/package-task.rb#L140

BTW, we have `.gitignore` for `dev/tasks/linux-packages/`: https://github.com/apache/arrow/blob/main/dev/tasks/linux-packages/.gitignore

Reply:
> Hmm. This directory isn't created by default.
> Did you use `BUILD_DIR=build`?

No, I didn't.

Reply:
Did you copy & paste this?
https://github.com/apache/arrow/blob/4565b613a488516b9bcbc664ad21d92051ea4fe6/dev/tasks/linux-packages/github.linux.yml#L68

Reply:
Ah, I probably did at the some point, since I was trying to reproduce those builds :)

Reply:
OK. Can we close this PR?

Reply:
Well, isn't it common to do just what I did? It would still sound useful to have a gitignore pattern for that, we already have other convenience patterns, see 
https://github.com/apache/arrow/blob/main/.gitignore and https://github.com/apache/arrow/blob/main/cpp/ for examples (text editor files, etc.).

Reply:
I added it for ccache in CI.

If we want to add it, could you do it in `dev/tasks/linux-packages/.gitignore` that already has entries for `dev/tasks/linux-packages/`?

```diff
diff --git a/dev/tasks/linux-packages/.gitignore b/dev/tasks/linux-packages/.gitignore
index 0e49a90c1e..138662a4ef 100644
--- a/dev/tasks/linux-packages/.gitignore
+++ b/dev/tasks/linux-packages/.gitignore
@@ -16,13 +16,15 @@
 # under the License.
 
 /*/*.tar.gz
-/*/apt/repositories/
-/*/apt/tmp/
 /*/apt/build.sh
+/*/apt/build/
 /*/apt/env.sh
-/*/yum/repositories/
-/*/yum/tmp/
+/*/apt/repositories/
+/*/apt/tmp/
 /*/yum/build.sh
+/*/yum/build/
 /*/yum/env.sh
+/*/yum/repositories/
+/*/yum/tmp/
 /apt/repositories/
```

Reply:
Suggestion applied.